tests/urlpatterns/polymorphic_urls.py

```diff
@@ -0,0 +1,32 @@
+from django.urls import path, re_path, register_converter
+
+from . import converters, views
+
+register_converter(converters.DynamicConverter, 'dynamic')
+
+
+# test reverse resolution of routes that share the same name, based on ...
+urlpatterns = [
+    # ... count
+    path('a/1/', views.empty_view, name='a'),
+    path('a/2/<value>/', views.empty_view, name='a'),
```

Comment:
Can we use more specific names?, e.g.

- `a` -> `number_of_kwargs`,
- `b` -> `kwarg`,
- `c` -> `converter`,
- `d` -> `regex`, etc.